source/Mlos.Python/mlos/Spaces/HypergridAdapters/unit_tests/TestCategoricalToOneHotEncodedHypergridAdapter.py

```diff
@@ -0,0 +1,173 @@
+
+    def test_projecting_dataframe_from_hierarchical_to_one_hot_encoding_cross_product_hypergrid(self):
+        adaptee = self.hierarchical_hypergrid
+        adapter = CategoricalToOneHotEncodedHypergridAdapter(adaptee=adaptee, merge_all_categorical_dimensions=True)
+        self._test_projecting_dataframe_categorical_to_one_hot_encoding_point_from_adaptee(adapter=adapter, adaptee=adaptee, num_random_points=1000)
+
+    def test_projecting_dataframe_from_hierarchical_to_one_hot_encoding_cross_product_drop_first_hypergrid(self):
+        adaptee = self.hierarchical_hypergrid
+        adapter = CategoricalToOneHotEncodedHypergridAdapter(adaptee=adaptee, merge_all_categorical_dimensions=True, drop='first')
+        self._test_projecting_dataframe_categorical_to_one_hot_encoding_point_from_adaptee(adapter=adapter, adaptee=adaptee, num_random_points=1000)
+
+    # Helper functions
+    def _test_projecting_dataframe_categorical_to_one_hot_encoding_point_from_adaptee(self, adaptee, adapter, num_random_points: int):
+        original_df = adaptee.random_dataframe(num_samples=num_random_points)
+        projected_df = adapter.project_dataframe(df=original_df, in_place=False)
+        self.assertTrue(id(original_df) != id(projected_df))
+        for column in adapter.get_one_hot_encoded_column_names():
+            self.assertTrue(projected_df[column].between(0, 1).all())
+        unprojected_df = adapter.unproject_dataframe(df=projected_df, in_place=False)
+        self.assertTrue(original_df.equals(unprojected_df))
+
+        # Let's make sure that projecting in place works as expected.
+        projected_in_place_df = adapter.project_dataframe(original_df, in_place=True)
+        self.assertTrue(id(original_df) == id(projected_in_place_df))
+        self.assertTrue(projected_in_place_df.equals(projected_df))
+        unprojected_in_place_df = adapter.unproject_dataframe(projected_in_place_df, in_place=True)
+        self.assertTrue(id(original_df) == id(unprojected_in_place_df))
+        self.assertTrue(unprojected_in_place_df.equals(unprojected_df[unprojected_in_place_df.columns.values]))
+
+    def _test_projecting_categorical_to_one_hot_encoding_point_from_adaptee(self, adaptee, adapter, num_random_points: int):
+        # First make sure that none of the resulting dimensions are categorical.
+        #
+        self.assertFalse(any(isinstance(dimension, CategoricalDimension) for dimension in adapter.dimensions))
+
+        # Now let's make sure that we can translate a bunch of points successfully.
+        #
+
+        for _ in range(num_random_points):
+            original_point = adaptee.random()
+            projected_point = adapter.project_point(original_point)
+            # Now let's ascertain that each coordinate in the projected point is in fact a number.
+            #
+            self.assertTrue(all(isinstance(dim_value, Number) for dim_name, dim_value in projected_point))

```

Comment:
Could we add a check to make sure that:
1) all categorical dimensions were dropped,
2) all dummy_variables are in (0,1)?